perf/perf_pmu.py

```diff
@@ -209,6 +227,40 @@ def test_cpu_event_count(self):
         # cpu events count
         self._check_count('cpu')
 
+    def test_sysfs_events(self):
+        devices_events = ['cpu',  'hv_24x7', 'hv_gpci']
+        for type_events in devices_events:
+            directory_base = '/sys/bus/event_source/devices'
+            directory = os.path.join(directory_base, type_events, 'events')
+            print(directory)
+            random_files = self.get_random_filenames(directory)
+            self._create_temp_user()
+            if not process.system('id test_pmu', sudo=True, ignore_status=True):
+                for file in random_files:
+                    eventdir = os.path.join(directory, file)
+                    result = process.run("su - test_pmu -c 'echo 1 >  %s'" % eventdir,
+                                         shell=True, ignore_status=True)
+                    output = result.stdout.decode() + result.stderr.decode()
+                    self.log.info("Test passes for %s of %s" %
+                                  (file, type_events))
+                if 'Permission denied' not in output:
+                    self.fail("Able to read  %s of %s as normal user" %
+                              (file, type_events))
+                self._remove_temp_user()
+            else:
+                self.log.warn('User test_pmu does not exist, skipping test')
+
+    def test_checkPowerVersion(self):
+        modes = ['POWER8', 'POWER9', 'POWER10']
+        proc_cpuinfo_path = "/proc/cpuinfo"
+        compat_mode = self.check_ProcessorCompatMode(proc_cpuinfo_path, modes)
+        cmd = "cat /sys/bus/event_source/devices/cpu/caps/pmu_name"
+        sysfs_value = process.system_output(cmd, shell=True,
+                                            ignore_status=True).decode()
+        if not compat_mode == sysfs_value:
+            self.fail(
+                "Sysfs cpu caps value differs from cpu compat mode in /proc/cpuinfo")
```

Comment:
This was a feature to add caps directory to get registered pmu information.
Please rename where-ever needed and remove the comparison part since for lpar booted in compat mode both values will not be same.
```
# cat /sys/bus/event_source/devices/cpu/caps/pmu_name
Power11
# tail /proc/cpuinfo
cpu		: POWER9 (architected), altivec supported
```